Plugins/Flow.Launcher.Plugin.Program/Main.cs

```diff
@@ -60,15 +60,26 @@ public async Task<List<Result>> QueryAsync(Query query, CancellationToken token)
             var result = await cache.GetOrCreateAsync(query.Search, async entry =>
             {
                 var resultList = await Task.Run(() =>
-                    _win32s.Cast<IProgram>()
-                        .Concat(_uwps)
-                        .AsParallel()
-                        .WithCancellation(token)
-                        .Where(HideUninstallersFilter)
-                        .Where(p => p.Enabled)
-                        .Select(p => p.Result(query.Search, Context.API))
-                        .Where(r => r?.Score > 0)
-                        .ToList());
+                {
+                    try
+                    {
+                        return _win32s.Cast<IProgram>()
+                            .Concat(_uwps)
+                            .AsParallel()
+                            .WithCancellation(token)
+                            .Where(HideUninstallersFilter)
+                            .Where(p => p.Enabled)
+                            .Select(p => p.Result(query.Search, Context.API))
+                            .Where(r => r?.Score > 0)
+                            .ToList();
+                    }
+                    catch (OperationCanceledException)
+                    {
+                        // Fix #3091: System.OperationCanceledException Issue
```

Comment:
@Jack251970 please remove this comment and link the issue to PR instead

Reply:
Please also add debug logging.

Reply:
> What happens to flow UI/UX wise when this exception is caught? If the result is empty, we need to consider error logging so it's not swallowed.

The result should be non-empty because other QueryAsync functions should be called. Considering some unknown situations, I added the warning log output.

Reply:
It should only affect debug code. This error should be caught outside. Though it can be a little bit annoying when debugging though.

Reply:
Get it.

Reply:
> It should only affect debug code. This error should be caught outside. Though it can be a little bit annoying when debugging though.

It's caught outside but iirc it is not descriptive enough from the logs to figure out what is causing the error (or maybe it is and I just confuse it with another issue, because I see operation cancelled issue raised more times than usual).

@Jack251970 @taooceros thoughts about logging as error for now so we can see from users how often this happens?

Reply:
>It should only affect debug code
Actually @taooceros what do you mean it only affects debug code? Does this not happen in production?

Reply:
Please refere to (#3091) <img width="1280" alt="image" src="https://github.com/user-attachments/assets/01e5b124-9432-446f-a0dc-da6d0dcac3aa">.
This issue is captured under VS in debug mode.

It seems there are try catch statements outside the codes in production, so this exception can be handled outside in production, while in debug not (pop up a exception like that screenshot).
Thats why @taooceros says 'Though it can be a little bit annoying when debugging though.'.

Also, I think handling exceptions in the inner try-catch block can improve performance in certain scenarios.

Reply:
> > It should only affect debug code
> 
> Actually @taooceros what do you mean it only affects debug code? Does this not happen in production?

I think @taooceros perhaps mean this.

Reply:
> > It should only affect debug code. This error should be caught outside. Though it can be a little bit annoying when debugging though.
> 
> It's caught outside but iirc it is not descriptive enough from the logs to figure out what is causing the error (or maybe it is and I just confuse it with another issue, because I see operation cancelled issue raised more times than usual).
> 
> @Jack251970 @taooceros thoughts about logging as error for now so we can see from users how often this happens?

It should only be noticeable when you are debugging flow with a debugger.

Reply:
Sure I agree we can do that. This can be quite annoying since the program plugin often encounter cancellation

Reply:
Hmm ok in that case let's keep as debug logging then.